### PR TITLE
Pr etiquette

### PR DESCRIPTION
**Pull request by ingkebil**
Adds a simple write up on how to lower expectations.

How to view changes made to manual?

```
git clone -b <branch name> https://github.com/Clinical-Genomics/development.git
cd development
conda create -n D_mkdocs
source activate D_mkdocs
pip install -r requirements.txt
mkdocs serve
```

go to your browser: http://127.0.0.1:8000

### CODE REVIEW

**Comment by barrystokman** [comment]
Couple of remarks:

> Most teams have a daily stand up. Why not join them at that time?

A stand up is for the Scrum team. While anyone that is not part of the team is free and even encouraged to attend, they are not allowed to participate. No need to change the text, I will handle that kind of stuff when it turns up by suggesting we look at the sprint board first and at the PR board second. This was just FYI.

> In other cases, it is recommended to ask the target audience to test the feature

In the case of functional testing, the target audience would be the development team, and not a TTN member. Should we make that more clear?

**Comment by ingkebil** [comment]
> Couple of remarks:
> 
> > Most teams have a daily stand up. Why not join them at that time?
> 
> A stand up is for the Scrum team. While anyone that is not part of the team is free and even encouraged to attend, they are not allowed to participate. No need to change the text, I will handle that kind of stuff when it turns up by suggesting we look at the sprint board first and at the PR board second. This was just FYI.

The idea here was that no one is continuously interrupted by people coming to their desks asking if a PR was seen. I wanted to group the requests a bit, if possible. I understand that using the daily standup is misleading and that it is very scrum specific. I can just say "after the daily standup" ;)

> > In other cases, it is recommended to ask the target audience to test the feature
> 
> In the case of functional testing, the target audience would be the development team, and not a TTN member. Should we make that more clear?

**Comment by ingkebil** [comment]
> > In other cases, it is recommended to ask the target audience to test the feature
> 
> In the case of functional testing, the target audience would be the development team, and not a TTN member. Should we make that more clear?

Yes, good point. Totally did not cover that angle.

So the target audience would be someone that would be using the feature. e.g. for a change in the order portal, we usually ask someone of the lab to test it out. I'll have to think about how to formulate this so that we don't have to mention team names.

**Comment by ingkebil** [comment]
OK, I made the part about testing less explicit by formulating it in such way that you should be specific in what you're asking of the reviewer.

File: docs/github/pr-etiquette.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+![that man has a place to be](dilbert-pr.png)
+
+What exactly should one expect when requesting or reviewing a pull request?
+
+As a requester:
+
+ - [Request a review](pr-request.md) to the code owner of the tool, not a team. This should happen automatically when [code owners](https://help.github.com/en/articles/about-code-owners) are set up for the repository.
+ - Reach out within the pull request or preferably face to face with the reviewers to communicate the urgency and get an estimated time to review. Most teams have a daily standup. Why not join them after that time?
+ - Be specific about what you're asking from the reviewer: are you asking for a code review or to test your pull request?
```
Comment (ingkebil) [inline docs/github/pr-etiquette.md]: Might add the sentence: "It is OK albeit not recommended, to test your own pull request."

**Comment by ingkebil** [comment]
Do you approve of those changes @barrystokman ?

**Comment by barrystokman** [comment]
Yes, but in case PRs need to be reviewed by TTN, they should already be tested. That needs to be clear.

And if you want to take some of the testing away from the developer to avoid 'de slager die zijn eigen vlees keurt' scenarios, don't let them **specify** test cases. Test execution is just pushing buttons, proper test coverage is determined when you create the test cases.

> The developer knows how their code works and will fall into the habit of testing their code according to this knowledge.
>
> The developer will find it difficult to remove themself from the mindset of 'how it works' as opposed to 'how it should work'.
>
> Because of this it is better to get someone with a high degree of objectivity to test the program i.e QA or Test Engineers

[source](https://softwareengineering.stackexchange.com/questions/77146/why-to-let-not-let-developers-test-their-own-workl)

**Comment by ingkebil** [comment]
Alright. I understand the explanation about why a developer shouldn't test their own code. That explanation does not belong in a pr etiquette post, which should only highlight some behavioral tips in regards to expectations.

The explanation on who should test should deserve its own section. I will summarize what we have gathered so far and send it for review.

**Pull request merged**